library/blivet.py

```diff
@@ -609,12 +611,12 @@ def _create(self):
 
         fmt = self._get_format()
 
-        create_vdo = self._blivet_pool._pool['deduplication'] or self._blivet_pool._pool['compression']
+        create_vdo = self._volume['deduplication'] or self._volume['compression']
 
         if create_vdo:
-            pool_size = size
-            if self._blivet_pool._pool['vdo_pool_size']:
-                pool_size = Size(self._blivet_pool._pool['vdo_pool_size'])
+            pool_size = parent.free_space
+            if self._volume['vdo_pool_size']:
```

Comment:
I thought the vdo pool size was going to default to matching the volume size. Here it defaults instead to taking all the free space in the pool/vg.

Reply:
Fixed. This needed to be updated. With VDO, `size` option is allowed to be larger than available space in pool.
Default value is now `size` or `parent.free_space` whichever is lesser.